From dfe434e66d3c9e11df322be62280f4ef0e823cc1 Mon Sep 17 00:00:00 2001
Message-Id: <dfe434e66d3c9e11df322be62280f4ef0e823cc1.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: "Kumar, Kiran S" <kiran.s.kumar@intel.com>
Date: Wed, 28 Jan 2015 14:34:57 +0530
Subject: [PATCH 02/95] FOR_UPSTREAM [VPG]: drm/i915 Remove DL programming for
 sprite planes during pixel format change.

As DL masking is not used, delete these. And no need to set true for
last plane state while plane update.

Issue: GMINL-5637
Change-Id: I8a7e5cd188c56af426e39c65eb4e756d2f8bc3cf
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.h     |    3 ---
 drivers/gpu/drm/i915/intel_drv.h    |    1 -
 drivers/gpu/drm/i915/intel_sprite.c |    2 --
 3 files changed, 6 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.h b/drivers/gpu/drm/i915/i915_drv.h
index d0bbc74..30f4bb8 100644
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -1512,9 +1512,6 @@ struct i915_dpst_registers {
 	uint32_t blm_hist_bin_count_mask;
 };
 
-#define DL_PRIMARY_MASK 0x000000ff
-#define DL_SPRITEA_MASK 0x0000ff00
-#define DL_SPRITEB_MASK 0x00ff0000
 #define VLV_PLANES_PER_PIPE 4
 #define VLV_UPDATEPLANE_STAT_PRIM_PER_PIPE(pipe) \
 	(1 << (VLV_PLANES_PER_PIPE * (pipe)))
diff --git a/drivers/gpu/drm/i915/intel_drv.h b/drivers/gpu/drm/i915/intel_drv.h
index 23c1eb7..0d368d0 100644
--- a/drivers/gpu/drm/i915/intel_drv.h
+++ b/drivers/gpu/drm/i915/intel_drv.h
@@ -462,7 +462,6 @@ struct intel_crtc {
 	bool primary_alpha;
 	bool sprite0_alpha;
 	bool sprite1_alpha;
-	uint32_t last_pixel_size;
 
 	/* reset counter value when the last flip was submitted */
 	unsigned int reset_counter;
diff --git a/drivers/gpu/drm/i915/intel_sprite.c b/drivers/gpu/drm/i915/intel_sprite.c
index a9b71df..4470864 100644
--- a/drivers/gpu/drm/i915/intel_sprite.c
+++ b/drivers/gpu/drm/i915/intel_sprite.c
@@ -696,8 +696,6 @@ vlv_update_plane(struct drm_plane *dplane, struct drm_crtc *crtc,
 
 	dev_priv->plane_stat |= VLV_UPDATEPLANE_STAT_SP_PER_PIPE(pipe, plane);
 
-	intel_plane->last_plane_state = INTEL_PLANE_STATE_ENABLED;
-
 	if (!dev_priv->atomic_update)
 		intel_flush_primary_plane(dev_priv, intel_crtc->plane);
 
-- 
1.7.9.5

